### PR TITLE
my mods to Jgfouca/optimize preview namelists

### PR DESCRIPTION
**Pull request by jedwards4b**
[ Description of the changes in this Pull Request. It should be enough
information for someone not following this development to understand.
Lines should be wrapped at about 72 characters. ]

Test suite: 
Test baseline: 
Test namelist changes: 
Test status: [bit for bit, roundoff, climate changing]

Fixes [CIME Github issue #]

User interface changes?: 

Code review:

### CODE REVIEW

**Comment by jgfouca** [comment]
All ready?

**Comment by rljacob** [comment]
Please fix the PR description and title.

File: utils/python/CIME/build.py [hunk 1]
```diff
@@ -396,6 +397,10 @@ def build_checks(case, build_threaded, comp_interface, use_esmf_lib, debug, comp
     case.set_value("BUILD_COMPLETE", False)
 
     case.flush()
+    if not model_only:
```
Comment (jgfouca) [inline utils/python/CIME/build.py]: This looks different than the call that was there before, is that on purpose?

Comment (jedwards4b) [inline utils/python/CIME/build.py]: Yes, I realized that this needs to happen before the sharedlib build not after

Comment (jgfouca) [inline utils/python/CIME/build.py]: OK, thanks

**Comment by jedwards4b** [comment]
ready

**Pull request merged**